roles/jd-client/src/lib/upstream_sv2/upstream.rs

```diff
@@ -231,13 +231,14 @@ impl Upstream {
         };
 
         // Gets the binary frame message type from the message header
-        let message_type = if let Some(header) = incoming.get_header() {
-            header.msg_type()
-        } else {
-            return Err(framing_sv2::Error::ExpectedHandshakeFrame.into());
-        };
+        let message_type = incoming.header().msg_type();
         // Gets the message payload
-        let payload = incoming.payload();
+        let payload = match incoming.payload() {
```

Comment:
This seems like a place where clippy would complain to use if let Some . But if clippy is happy I'm as well. Maybe since you are at it you could just remove the above comment.